### ISSUE TITLE
Mixed JSON format

### ISSUE THREAD

**Issue by chrisgaun** [solo-io/gloo#5689]
The WAF space delimited logs are appended to end of JSON which makes it hard to parse in centralize logging tools like Datadog. 

Would like to use JSON logs instead of space delimited.

**Comment by ben-taussig-solo** [comment]
To me it seems likely that this is caused by an improperly specified formatting string. For example, the following format string will format JSON values in the modsecurity audit log properly:

```yaml
...
jsonFormat:
  waf: "%DYNAMIC_METADATA(io.solo.filters.http.modsecurity:audit_log)%"
```

whereas this format string (note the stray `\n`):

```yaml
...
jsonFormat:
  waf: "%DYNAMIC_METADATA(io.solo.filters.http.modsecurity:audit_log)%\n"
```

will mangle JSON formatted audit logs.

Do we have more information about the configuration that was used when this issue came up?

**Comment by ben-taussig-solo** [comment]
This issue seems to be a duplicate of this now-closed issue: https://github.com/solo-io/gloo/issues/6141

That issue was resolved with the following PR, which clarifies the proper use of access logging format strings in our documentation, in the same manner as the comment above: https://github.com/solo-io/gloo/pull/6350

**Comment by oblogic7** [comment]
We are handling this issue via Datadog pipeline using the following grok parser rule. The rule may need to be updated depending on which fields are included in your logs and the json needs to appear at the end of the string for it to work since the `?%{data::json}` field will match all remaining text in the log entry.

```
access_log \[%{date("yyyy-MM-dd'T'HH:mm:ss.SSSZ"):date}\] "%{word:http.method} %{notSpace:http.url} %{notSpace:http.protocol}" %{integer:http.status_code} %{data:response.flags} %{integer:network.bytes_read} %{integer:network.bytes_written} %{integer:duration:scale(1000000)} %{notSpace:response.x_envoy_upstream_service_time} "%{data:http._x_forwarded_for}" "%{data:http.useragent}" "%{notSpace:http.request_id}" "%{notSpace:http.authority}" %{doubleQuotedString:upstream_host} ?%{data::json}

```

**Comment by nfuden** [comment]
Closing as we seem to agree that between following the specific mod-security formatting issues  and guides for postprocessing as seen with the above comment
